R/probability_distributions.R

```diff
@@ -22,55 +23,32 @@ uniform_distribution <- R6Class (
 
       }
 
-      if (!is.finite(min) | !is.finite(max)) {
+      if (!all(is.finite(min)) | !all(is.finite(max))) {
 
         stop ('min and max must finite scalars',
               call. = FALSE)
 
       }
 
-      if (min >= max) {
+      if (any(min >= max)) {
 
         stop ('max must be greater than min',
               call. = FALSE)
 
       }
 
-      # store min and max as numeric scalars (needed in create_target, done in
-      # initialisation)
-      self$min <- min
-      self$max <- max
-
-      self$bounds <- c(min, max)
-
       # initialize the rest
       super$initialize('uniform', dim)
 
       # add them as children and greta arrays
       self$add_parameter(min, 'min')
       self$add_parameter(max, 'max')
-
-      # the density is fixed, so calculate it now
-      self$log_density <- -log(max - min)
-
-    },
-
-    # default value (ignore any truncation arguments)
-    create_target = function (...) {
-      vble(truncation = c(self$min, self$max),
-           dim = self$dim)
+      
     },
 
     tf_distrib = function (parameters) {
-
-      tf_ld <- fl(self$log_density)
-
-      # weird hack to make TF see a gradient here
-      log_prob = function (x)
-        tf_ld + x * fl(0)
-
-      list(log_prob = log_prob, cdf = NULL, log_cdf = NULL)
-
+      tf$contrib$distributions$Uniform(low = parameters$min,
```

Comment:
Hrmmm. Using this distribution (rather than pre-calculating the density as a constant) will probably be less performant. Though perhaps not noticably so, and maybe tensorflow can skip recalculating it. If it works for you, than go for it.

Could you please add to the [density tests for uniform](https://github.com/greta-dev/greta/blob/release_candidate_0.2.5/tests/testthat/test_distributions.R#L17-L29), so that it's tested with vector-valued min and max?

`uniform()` is a bit weird, since it needs fixed data to be able to have a reasonable transformation to unconstrained space. There might be a better way of solving that problem however. One idea would be to handle a `uniform(0, 1)` variable internally, and then rescale using greta arrays for min and max, with the `representations` mechanism in greta. That's a separate issue though.